### ISSUE TITLE
Re-write a/v METS to use file groups with usage attributes

### ISSUE THREAD

**Issue by aray-wellcome** [wellcomecollection/goobi-infrastructure#308]
We started our a/v workflow overhaul with writing a simple METS file for the films and relying on DDS to map what to do with each file based on the MIMETYPE. 

We want to be able to add transcripts to our workflow, though, before we deem it complete. This has caused some discussion among us because it makes less sense to have a rule hard coded into DDS that handles a film + transcript + post image combination.  

So it has been suggested by Tom Crane that we add more information in the METS so DDS doesn't need the hard coded rule and can be more flexible. 

A snipped of the fileGrp of our current film METS:

```xml
<mets:fileSec>
    <mets:fileGrp USE="OBJECTS">
      <mets:file ID="FILE_0001_OBJECTS" MIMETYPE="image/jpeg">
        <mets:FLocat LOCTYPE="URL" xlink:href="objects/b32162790_0001.jpg" />
      </mets:file>
      <mets:file ID="FILE_0002_OBJECTS" MIMETYPE="video/mp4">
        <mets:FLocat LOCTYPE="URL" xlink:href="objects/b32162790_0002.mp4" />
      </mets:file>
      <mets:file ID="FILE_0003_OBJECTS" MIMETYPE="application/mxf">
        <mets:FLocat LOCTYPE="URL" xlink:href="objects/b32162790_0003.mxf" />
      </mets:file>
    </mets:fileGrp>
  </mets:fileSec>
```

After discussing with Tom Crane and Jonathan Tweed, it seems that we'd like to perhaps move forward with our film and audio METS being written with specific file groups and usage attributes much like we see with the British Library's Sound Collection METS:
https://sounds.bl.uk/Environment/British-wildlife-recordings/022M-W1CDR0001531-2100V0

```xml
<mets:fileSec>
<mets:fileGrp ID="ASRprojectdocuments">
<mets:file ID="BLAPSoundASR2-v0-5">
<mets:FLocat LOCTYPE="URL" xlink:href="http://www.bl.uk/profiles/sound/BLAPSoundASR2-v0-5"/>
</mets:file>
<mets:file ID="Blapsi_v1.3.pdf">
<mets:FLocat LOCTYPE="URL" xlink:href="http://sounds.bl.ul/Blapsi_v1.3.pdf"/>
</mets:file>
<mets:file ID="UID_Sound_v.9.pdf">
<mets:FLocat LOCTYPE="URL" xlink:href="http://sounds.bl.ul/UID_Sound_v.9.pdf"/>
</mets:file>
</mets:fileGrp>
<mets:fileGrp ID="MASTERS" USE="PRESERVATION">
<mets:file ID="FILE_001" MIMETYPE="sound/wav" USE="preservation master" AMDID="techMD001">
<mets:FLocat LOCTYPE="URL" xlink:href="sounds.bl.uk/022A-W1CDR0001531-2100M0.wav"/>
</mets:file>
</mets:fileGrp>
<mets:fileGrp ID="DERIVATIVES" USE="PLAY BACK AND ACCESS">
<mets:file ID="FILE_002" MIMETYPE="sound/wav" USE="play back" AMDID="techMD002">
<mets:FLocat LOCTYPE="URL" xlink:href="sounds.bl.uk/022A-W1CDR0001531-2100P0.wav"/>
</mets:file>
<mets:file ID="FILE_003" MIMETYPE="sound/wma" USE="access (stream)" AMDID="techMD003">
<mets:FLocat LOCTYPE="URL" xlink:href="sounds.bl.uk/022A-W1CDR0001531-2100A0.wma"/>
</mets:file>
<mets:file ID="FILE_004" MIMETYPE="sound/mp3" USE="access (download)" AMDID="techMD004">
<mets:FLocat LOCTYPE="URL" xlink:href="sounds.bl.uk/022A-W1CDR0001531-2100A0.mp3"/>
</mets:file>
</mets:fileGrp>
</mets:fileSec>
```

The fileGrps and usage attributes we might use, for example, could be:
1.  mets:fileGrp ID= "MASTERS" USE="PRESERVATION"
The MXF file would go in this group
2, mets:fileGrp ID= "OBJECTS" USE="ACCESS"
The MPG or MP4 would go in here (or the .wav or .mp3 for audio)
3. mets:fileGrp ID= "POSTER" USE="POSTER IMAGE"
The jpg poster image would go here
4. mets:fileGrp ID= "TRANSCRIPT" USE="TRANSCRIPT"
The PDF or (in the future, html transcript) would go here

These filegroup names and use attributes can be renamed of course if there's better suggestions. 

So, these are our initial thoughts on the METS for the a/v workflows. Does this sound like something Goobi can do for us? Is it logical? Do you have any other advice or suggestions?

**Comment by rsehr** [comment]
This was one of my proposals in this issue: #255 

Compared to the current solution, it has the advantage that it can be extended flexibly and the different formats are treated as belonging together as different manifestations of the same physical object.

However, it also has a major disadvantage, namely that there can only be one amdSec for the physical object, which means that we can only store one premis information. The premis information refers to the object in the structMap and not to the fileGrp, so it will be difficult to store the data for all manifestations. Maybe we can solve it by creating a direct link between amdSec/techMD and fileGrp/file/FLocat.

btw the link to BL is a bad example. There you can find the following in the METS file:
  <mets:structMap TYPE="physical" />
  <mets:structMap TYPE="logical" />
  <mets:structLink />

So this object is actually non-existent and the various fileGrp do not belong anywhere.

**Comment by aray-wellcome** [comment]
I'm sure it'll come to no surprise to you that I'm still learning about METS so it took me a while to understand what this was saying! And I think I understand it 70%...

Would it be possible to see if we can solve the issue by creating that direct link? I'd be interested to see what it looks like, I always learn better with examples in front of me. 

Or do you have any other suggestions about what we could do maybe?

**Comment by rsehr** [comment]
I have added 3 files. They are METS xml, but must be named txt to be able to upload them. The file b30495805_current.txt is in principle the current solution extended by the PDF.

b30495805_one_object.txt contains a single element in the physical structMap. However, it refers to 4 different manifestations. Each manifestation is stored in its own fileGrp and linked to the premis data via the ADMID.

b30495805_combination.txt contains a combination of the other two. For each existing file there is an equivalent in the structMap, the premis data is linked via the ADMID. However, there is now a separate fileGrp for each format, which contains only one file.

I would prefer the solution from 30495805_one_object.txt as it most clearly expresses what kind of object is described here. However, it also means the biggest change. In Goobi, we have to work with different folders for the different formats, the export of the METS file and the linking of the Premis data must also be adapted.

b30495805_combination.txt would be less complex to implement, here we would only have to make sure that a separate fileGrp is created for each format. But the mets file would suggest that we have a sequence of 4 different objects.

[b30495805_combination.txt](https://github.com/wellcomecollection/goobi-infrastructure/files/4786549/b30495805_combination.txt)
[b30495805_current.txt](https://github.com/wellcomecollection/goobi-infrastructure/files/4786551/b30495805_current.txt)
[b30495805_one_object.txt](https://github.com/wellcomecollection/goobi-infrastructure/files/4786552/b30495805_one_object.txt)

**Comment by tomcrane** [comment]
_b30495805_one_object_ looks the clearest model of the object for me too, and allows for future flexibility. In the example, the physical structMap is a sequence with one element, that contains 4 files, each of which is a different aspect of the Video. Using this approach, an object that comprised two videos (maybe 2 reels of film separately digitised) would have two elements in the sequence, each with their four file aspects. The meaning of the sequence doesn't get confused by different formats/uses appearing as if they were part of the sequence, which is good.

That said, it's also the biggest change for DDS. The ADMID attribute moves from the div in the physical structMap to the file element in the fileGrp. I think I would like to make a few changes to the dashboard UI to better indicate what is going on here, when visualising what kind of thing this is.

Both _b30495805_combination_ and _b30495805_one_object_ require changes to the DDS, just as they do to Goobi. _b30495805_one_object_ requires more changes, but if we're changing the DDS anyway we might as well get it right. The DDS would need to look in more places for the links between parts of the METS file, and handle a bit more variation, but that's OK - I think the needs of the DDS are subservient to the source-of-truth for the model of the digital object, which is the METS file, so all other things being equal, that should be the thing that is most expressive of what the object is, and the DDS adapt to it.

In summary - the DDS can work with either of these two models, so it's your choice. I agree with @rsehr that _b30495805_one_object_ is a better model.

**Comment by aray-wellcome** [comment]
It seems that b30495805_one_object is the METs model we should move forward with so happy to get that work started.

**Comment by rsehr** [comment]
- [ ] allow upload of pdf transcripts
- [ ] move different formats in seperate folder
- [ ] create pagination/physical structure based on data from main folder
- [ ] create premis metadata for all files from different folder
- [ ] create separate filegroups for each format/folder
- [ ] link file in filegroup to premis metadata
- [ ] allow new filegrpoups in metadata validation for bagit creation
- [ ] add files from all folder to bag
- [ ] create manifest for all files

**Comment by rsehr** [comment]
Changes:

configure separate folder for poster and transcript files in goobi_config.properties:

process.folder.images.poster={processtitle}_poster
process.folder.images.transcript={processtitle}_transcript

Configure different file groups for the Wellcome_Film project:

Name: OBJECTS
Path: master/
MIME type: video/mp4
Suffix: mp4
Validation against directory: 
Use file extensions and MIME type of files: yes

Name: POSTER
Path: master/
MIME type: image/jpg
Suffix: jpg
Validation against directory: poster
Use file extensions and MIME type of files: no

Name: MASTERS
Path: master/
MIME type: video/mxf
Suffix: mxf
Validation against directory: 
Use file extensions and MIME type of files: no

Name: TRANSCRIPT
Path: master/
MIME type: application/pdf
Suffix: pdf
Validation against directory: transcript
Use file extensions and MIME type of files: no

Add new task to the Wellcome_Film processes:
Title: Move files
Order: 25
Automatic task: true
Step plugin: intranda_step_movevideofiles
User group: AutomaticTasks

Change http JSON body in 'bagit creation and upload':

{
    "goobiId":"{processid}",
    "sourceDir":"",
    "targetDir":"",
    "fontType":"",
    "priority":5,
    "postOCR":"{stepid}",
    "titleprefix":"{processtitle}",
    "templatename":"bagit",
    "jobtype":"BAGIT",
    "serverType":"",
    "title":"{meta.TitleDocMain}",
    "bNumber":"{meta.CatalogIDDigital}",
    "goobiTitle":"{processtitle}",
    "goobiID":"{processid}",
    "metsFile":"{processpath}/{processtitle}.xml",
    "anchorFile":"{processpath}/{processtitle}_anchor.xml",
    "s3Prefixalto":"{s3_ocrbasispath}{processtitle}_alto",
    "outputBucket":"wellcomecollection-workflow-export-bagit-stage",
    "archiveType":"tar.gz",
    "video":"true",
    "s3PrefixImages":"{s3_tifpath}",
    "s3PrefixMaster":"{s3_origpath}",
    "s3PrefixPoster":"{s3_imagepath}{processtitle}_poster",
    "s3PrefixTranscript":"{s3_imagepath}{processtitle}_transcript"
}

**Comment by rsehr** [comment]
The installation on production is blocked until the new mets profile was implemented by digirati

**Comment by aray-wellcome** [comment]
Closing as the METS work has been completed in Goobi. Will reopen a new ticket if any adjustments need to be made when Digirati pick this up.
